packages/core/phase3/lib/pncUpdateRequestGenerators/remandGenerator/getResultsForRemand.test.ts

```diff
@@ -0,0 +1,212 @@
+              BottomLevelCode: "00",
+              OrganisationUnitCode: "A01CJ00"
+            }
+          : {
+              TopLevelCode: "A",
+              SecondLevelCode: "01",
+              ThirdLevelCode: "CJ",
+              BottomLevelCode: "00",
+              OrganisationUnitCode: "A01CJ00"
+            },
+      PNCDisposalType: options.recordableResult === false ? 1000 : 2051,
+      PNCAdjudicationExists: options.pncAdjudicationExists ?? true,
+      ResultClass: options.resultClass ?? ResultClass.ADJOURNMENT_POST_JUDGEMENT
+    }
+  ] as Result[]
+})
+
+const operation = {
+  code: "NEWREM",
+  data: {
+    nextHearingDate: new Date("2024-12-11T10:11:12.000Z"),
+    nextHearingLocation: {
+      TopLevelCode: "A",
+      SecondLevelCode: "01",
+      ThirdLevelCode: "CJ",
+      BottomLevelCode: "00",
+      OrganisationUnitCode: "A01CJ00"
+    }
+  }
+} as Operation<PncOperation.REMAND>
+
+describe("getResultsForRemand", () => {
+  it("should not return any results when offence is not recordable", () => {
+    const offences = [createOffence({ recordableOffence: false })] as Offence[]
+
+    const results = getResultsForRemand(offences, operation)
+
+    expect(results).toHaveLength(0)
+  })
+
+  it("should not return any results when result is not recordable", () => {
+    const offences = [createOffence({ recordableResult: false })] as Offence[]
+
+    const results = getResultsForRemand(offences, operation)
+
+    expect(results).toHaveLength(0)
+  })
+
+  it("should not return any results when next hearing date does not match the operation data", () => {
+    const offences = [createOffence({ nextHearingDateMatch: false })] as Offence[]
+
+    const results = getResultsForRemand(offences, operation)
+
+    expect(results).toHaveLength(0)
+  })
+
+  it("should not return any results when next result source organisation does match the operation data", () => {
+    const offences = [createOffence({ nextResultSourceOrganisation: false })] as Offence[]
+
+    const results = getResultsForRemand(offences, operation)
+
+    expect(results).toHaveLength(0)
+  })
+
+  it("should not return any results when PNC adjudication does not exist and result class is Adjournment post Judgement", () => {
+    const offences = [
+      createOffence({ pncAdjudicationExists: false, resultClass: ResultClass.ADJOURNMENT_POST_JUDGEMENT })
+    ] as Offence[]
+
+    const results = getResultsForRemand(offences, operation)
+
+    expect(results).toHaveLength(0)
+  })
+
+  it.each([ResultClass.ADJOURNMENT, ResultClass.ADJOURNMENT_PRE_JUDGEMENT])(
+    "should not return any results when PNC adjudication exists and result class %s",
+    (resultClass) => {
+      const offences = [createOffence({ pncAdjudicationExists: true, resultClass })] as Offence[]
+
+      const results = getResultsForRemand(offences, operation)
+
+      expect(results).toHaveLength(0)
+    }
+  )
+
+  it("should return the matching result", () => {
```

Comment:
Is it possible to make one of the results non-matching? As all the results match meaning we're kind of just returning the results from each offence.

Reply:
Would also be good to extract some variables for the expectation so it's clear which results are being returned from which offence. I struggled to tell what was being matched and from where.

Reply:
Good shout. Updated. One is not unmatching and it checks the object references for the other 2 results.